dependencies/dependencies.go

```diff
@@ -27,23 +29,31 @@ func (d DefaultDependencies) HTTPClient() (*http.Client, error) {
 	return nil, errors.New(codes.Invalid, "http client uninitialized in dependencies")
 }
 
+func (d DefaultDependencies) SecretService() (SecretService, error) {
+	if d.secretservice != nil {
+		return d.secretservice, nil
+	}
+	return nil, errors.New(codes.Invalid, "secret service uninitialized in dependencies")
+}
+
 // The values defined below come from the default implementation of Transport.
 // It establishes network connections as needed and caches them for reuse by subsequent calls.
 // It uses HTTP proxies as directed by the $HTTP_PROXY and $NO_PROXY (or $http_proxy and $no_proxy) environment variables.
 func NewDefaultDependencies() Interface {
```

Comment:
change to  

```suggestion
func NewDefaultDependencies() Interface {
  return &DefaultDependencies{
    httpclient: nil, 
    secretservice: nil,
    }
}
```

this is so that by default none of these dependencies are injected.